### ISSUE TITLE
1.12 `ENOENT: no such file or directory` referencing `outes` instead of `routes`

### ISSUE THREAD

**Issue by sockthedev** [remix-run/remix#5322]
### What version of Remix are you using?

1.12.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

- Create a Remix instance selecting the "Remix App Server" option
- Run `remix dev`

The following error occurs;

```
TypeError: Cannot read properties of undefined (reading 'remixRequestHandlerPath')
    at requestHandler (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/@remix-run+server-runtime@1.12.0/node_modules/@remix-run/server-runtime/dist/server.js:38:94)
    at /Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/@remix-run+express@1.12.0_express@4.18.2/node_modules/@remix-run/express/dist/server.js:39:28
    at /Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/@remix-run+serve@1.12.0/node_modules/@remix-run/serve/dist/index.js:47:7
    at Layer.handle [as handle_request] (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/express@4.18.2/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/express@4.18.2/node_modules/express/lib/router/route.js:144:13)
    at next (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/express@4.18.2/node_modules/express/lib/router/route.js:140:7)
    at next (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/express@4.18.2/node_modules/express/lib/router/route.js:140:7)
    at next (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/express@4.18.2/node_modules/express/lib/router/route.js:140:7)
    at next (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/express@4.18.2/node_modules/express/lib/router/route.js:140:7)
    at next (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/express@4.18.2/node_modules/express/lib/router/route.js:140:7)
```

### Expected Behavior

The dev server executes as before.

### Actual Behavior

An error occurs on dev server load.

**Comment by machour** [comment]
I'm not seeing this problem on my side, having just created a new project using `yarn create remix` and picking all the default options.

Could you detail the exact steps you're using to create the app?

**Comment by sockthedev** [comment]
I've tried to recreate with a new init and I can't seem to. But I can't fix my current instance either. I've deleted the package lock file and node modules, and can't see why or what is causing this issue.

Another strange issue. I can't build either;

```
Build failed with 1 error:
error: ENOENT: no such file or directory, open '/Users/sean/sockthedev/code/hivemind-tales/app/outes/_index/index.tsx'
Error
    at Object.onCompileFailure (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/@remix-run+dev@1.12.0_@r
emix-run+serve@1.12.0/node_modules/@remix-run/dev/dist/cli/commands.js:180:13)
    at Object.compile (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/@remix-run+dev@1.12.0_@remix-run+
serve@1.12.0/node_modules/@remix-run/dev/dist/compiler/remixCompiler.js:34:134)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.build (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/@remix-run+dev@1.12.0_@remix-
run+serve@1.12.0/node_modules/@remix-run/dev/dist/compiler/build.js:33:3)
    at async Object.build (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/@remix-run+dev@1.12.0_@remix-
run+serve@1.12.0/node_modules/@remix-run/dev/dist/cli/commands.js:175:3)
    at async Object.run (/Users/sean/sockthedev/code/hivemind-tales/node_modules/.pnpm/@remix-run+dev@1.12.0_@remix-ru
n+serve@1.12.0/node_modules/@remix-run/dev/dist/cli/run.js:449:7)
 ELIFECYCLE  Command failed with exit code 1.
ERROR: "build:remix" exited with 1.
 ELIFECYCLE  Command failed with exit code 1.
```

If you look at the log it looks like the `r` is missing from the `/routes/` in the filesystem path.

I've pushed up this repo. It's pretty close to vanilla Remix. All I've done is add basic tailwind config as per your docs, opted into v2 route convention, and created a couple of super simple components.

Are you able to run the following steps;

- git clone https://github.com/sockthedev/hive-mind-tales
- cd hive-mind-tales
- pnpm install
- pnpm build

**Comment by madispuk** [comment]
Same here. I'm using remix with express.
```
server-runtime/dist/server.js:38

// special __REMIX_ASSETS_MANIFEST endpoint for checking if app server serving up-to-date routes and assets
    let {
      unstable_dev
    } = build.future;
    if (mode$1 === "development" && unstable_dev !== false && url.pathname === (unstable_dev.remixRequestHandlerPath ?? "") + "/__REMIX_ASSETS_MANIFEST") {
```

Looks like this `build.future.unstable_dev` is undefined by default. So:
```
unstable_dev.remixRequestHandlerPath
```
should be:
```
unstable_dev?.remixRequestHandlerPath
```
or something. This PR:
https://github.com/remix-run/remix/pull/5243

**Comment by madispuk** [comment]
I had two dev dependencies which I forgot to update. This fixed the issue.
```
yarn add -D @remix-run/dev
yarn add -D @remix-run/eslint-config
```

**Comment by machour** [comment]
@sockthedev I'm 99% sure that you have an unaligned remix dependency, like @madispuk had.
Please check you package-lock.json file, and see if all remix packages are indeed on 1.12.0

You can do `grep @remix package-lock.json` to verify this

**Comment by sockthedev** [comment]
Hmmm, I was hopeful after that comment but the grep appears fine?

```
hivemind-tales on  main via  v16.19.0 took 2s 
❯ grep remix pnpm-lock.yaml
  '@remix-run/dev': ^1.12.0
  '@remix-run/eslint-config': ^1.12.0
  '@remix-run/node': ^1.12.0
  '@remix-run/react': ^1.12.0
  '@remix-run/serve': ^1.12.0
  '@remix-run/node': 1.12.0
  '@remix-run/react': 1.12.0_biqbaboplfbrettd7655fr4n2y
  '@remix-run/serve': 1.12.0
  '@remix-run/dev': 1.12.0_@remix-run+serve@1.12.0
  '@remix-run/eslint-config': 1.12.0_6phwajf43xwwgpkdgjffawkphy
  /@remix-run/dev/1.12.0_@remix-run+serve@1.12.0:
      '@remix-run/serve': ^1.12.0
      '@remix-run/serve':
      '@remix-run/serve': 1.12.0
      '@remix-run/server-runtime': 1.12.0
  /@remix-run/eslint-config/1.12.0_6phwajf43xwwgpkdgjffawkphy:
  /@remix-run/express/1.12.0_express@4.18.2:
      '@remix-run/node': 1.12.0
  /@remix-run/node/1.12.0:
      '@remix-run/server-runtime': 1.12.0
      '@remix-run/web-fetch': 4.3.2
      '@remix-run/web-file': 3.0.2
      '@remix-run/web-stream': 1.0.3
  /@remix-run/react/1.12.0_biqbaboplfbrettd7655fr4n2y:
      '@remix-run/router': 1.3.1
  /@remix-run/router/1.3.1:
  /@remix-run/serve/1.12.0:
      '@remix-run/express': 1.12.0_express@4.18.2
  /@remix-run/server-runtime/1.12.0:
      '@remix-run/router': 1.3.1
  /@remix-run/web-blob/3.0.4:
      '@remix-run/web-stream': 1.0.3
  /@remix-run/web-fetch/4.3.2:
      '@remix-run/web-blob': 3.0.4
      '@remix-run/web-form-data': 3.0.4
      '@remix-run/web-stream': 1.0.3
  /@remix-run/web-file/3.0.2:
      '@remix-run/web-blob': 3.0.4
  /@remix-run/web-form-data/3.0.4:
  /@remix-run/web-stream/1.0.3:
      '@remix-run/router': 1.3.1
      '@remix-run/router': 1.3.1
```

My package.json dependencies;

```json
{
  "dependencies": {
    "@remix-run/node": "^1.12.0",
    "@remix-run/react": "^1.12.0",
    "@remix-run/serve": "^1.12.0",
    "isbot": "^3.6.5",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@remix-run/dev": "^1.12.0",
    "@remix-run/eslint-config": "^1.12.0",
    "@types/react": "^18.0.25",
    "@types/react-dom": "^18.0.8",
    "eslint": "^8.27.0",
    "npm-run-all": "^4.1.5",
    "prettier": "^2.8.3",
    "tailwindcss": "^3.2.4",
    "typescript": "^4.9.5"
  }
}
```

**Comment by iamriot-dev** [comment]
I'm running into a similar issue. I upgraded an existing project to 1.12.0 and can no longer dev/build. It seems to be connected to the v2_routeConvention option.

With a fresh new project, the moment I turn on v2_routeConvention, I get the same error as my existing project:
```
✘ [ERROR] [plugin browser-route-module] ENOENT: no such file or directory, open '/Users/admin/projects/remix-test/app/outes/_index.tsx'
```
It will lists all the routes files correctly but with the wrong path. The build does complete when I rename the `routes` folder to `outes`, but loads blank pages when run.

**Comment by machour** [comment]
Shot in the dark: could you both add `@remix-run/server-runtime: ^1.12.0` in your dev dependencies and see if it fixes the issue ?

Would totally appreciate a public repository reproducing the issue

**Comment by sockthedev** [comment]
Shall try that, here is a public repository in the interim;

https://github.com/sockthedev/hive-mind-tales

**Comment by machour** [comment]
> Shall try that, here is a public repository in the interim;

Since I never used pnpm, would you mind updating the README with the actual commands to run using pnpm? 🙏🏼

**Comment by sockthedev** [comment]
Sure, I'll update it. The commands are;

```
pnpm install
pnpm build
```

**Comment by sockthedev** [comment]
I even tried some resolutions for good measure;

```json
{
  "resolutions": {
    "@remix-run/dev": "1.12.0",
    "@remix-run/eslint-config": "1.12.0",
    "@remix-run/node": "1.12.0",
    "@remix-run/react": "1.12.0",
    "@remix-run/serve": "1.12.0",
    "@remix-run/server-runtime": "1.12.0"
  }
}
```

But still get the following build error, similar to @iamriot-dev the "routes" path is being interpreted as `outes` by the compiler.

```
> @ build:remix /Users/sean/sockthedev/code/hivemind-tales
> remix build

Building Remix app in production mode...

✘ [ERROR] [plugin browser-route-module] ENOENT: no such file or directory, open '/Users/sean/sockthedev/code/hivemind-tales/app/outes/_index/index.tsx'
```

**Comment by iamriot-dev** [comment]
Here's the fresh project I was testing and having issues with: [https://github.com/iamriot-dev/test-remix](https://github.com/iamriot-dev/test-remix). For what it's worth, I'm using Yarn with PnP instead of pnpm.

By the way, you can see the misinterpretation of the route paths present itself in the `remix routes` cli command as well:
```
<Routes>
  <Route file="root.tsx">
    <Route path="index" file="outes/_index.tsx" />
  </Route>
</Routes>
```

**Comment by sapter** [comment]
> 

I've also just ran into this issue and it turned out it was due to unaligned remix dependencies.

**Comment by haines** [comment]
Is

```
TypeError: Cannot read properties of undefined (reading 'remixRequestHandlerPath')
```

related to 

```
error: ENOENT: no such file or directory, open '/Users/sean/sockthedev/code/hivemind-tales/app/outes/_index/index.tsx'
```

or are these two separate issues?

I have also raised the `outes` issue (https://github.com/remix-run/remix/issues/5324) and have pushed a fix (https://github.com/remix-run/remix/pull/5327). [Here's a workaround using patch-package](https://github.com/remix-run/remix/issues/5324#issuecomment-1410212098).

**Comment by sockthedev** [comment]
> or are these two separate issues?

I think the first error is a symptom of the issue when running the dev server.

**Comment by haines** [comment]
> I think the first error is a symptom of the issue when running the dev server.

Cool, in that case I've closed #5324 as a duplicate and linked #5327 as closing this issue instead 🙂

**Comment by sockthedev** [comment]
Awesome, thanks for jumping on this issue so fast!

**Comment by imprashast** [comment]
I can confirm the same behaviour. It trims the first word of our route, `npx remix routes` shows:

`<Route path="ndex" file="outes/index.tsx" />`

**Comment by adonaicandido** [comment]
I can confirm the same behavior as well. Updated an existing projeto to 1.12.0 and it doesn't dev/build.

Created a new project using indie stack, everything works great, expect if I try to use the v2 routes.

**Comment by gielcobben** [comment]
Same here

**Comment by MichaelDeBoey** [comment]
Closed by #5228
